tiledb/sm/query/query_condition.cc

```diff
@@ -1682,73 +1741,91 @@ void QueryCondition::apply_ast_node_sparse(
   }
 
   // Process the validity buffer now.
-  if (nullable) {
+  if (nullable && node->get_condition_value_view().content() == nullptr) {
```

Comment:
For apply_ast_node_sparse (we don't care too much about performance for apply_ast_node_dense), can we continue to process the nullable filed as it was before for AND conditions and use the nullable filed in the cell by cell comparisons for OR conditions?

Reply:
No, it won't be fully correct if we do that. 
The problem is that we should not count elements that pass the cmp check but are null in the array.
e.g. We want to do result_buffer[c] || (cmp & nullable_buffer[c]) but that result is not the same as (result_buffer[c] || cmp) & nullable_buffer[c])

Reply:
@abigalekim what I am saying is to make two code paths, one for AND that processes the query like it used to, and one for OR that does what you are implementing. The reason I'm asking is because we used to have the codebase do the null checks at the cell level, and because one customer complained about performance, we reorganized the code as it is today. I think your change will regress the performance for conditions that only have ANDs.